spec/02-integration/05-proxy/01-router_spec.lua

```diff
@@ -882,6 +882,119 @@ for _, strategy in helpers.each_strategy() do
       end)
     end)
 
+    describe("path prefixes and regexes", function()
+      local routes
+
+      lazy_setup(function()
+        routes = insert_routes {
+          {
+            strip_path = true,
+            paths      = { "/root/fix" },
+          },
+          {
+            strip_path = true,
+            paths      = { "/root/(fixture)" },
+          },
+          {
+            strip_path = true,
+            paths      = { "/root2/fixture" },
+          },
+          {
+            strip_path = true,
+            paths      = { "/root2/fix" },
+          },
+          {
+            strip_path = true,
+            paths      = { "/root2/(fixture)" },
+          }
+
+        }
+      end)
+
+      lazy_teardown(function()
+        remove_routes(routes)
+      end)
+
+      it("prioritizes regexes over prefixes", function()
```

Comment:
The documented Kong semantics is that it proritizes prefixes over regexes. See https://docs.konghq.com/1.0.x/proxy/#evaluation-order where it says:

> Prefix paths are always evaluated before regex paths.

Reply:
@hishamhm 

Please read:
https://github.com/Kong/kong/issues/4121
and
https://github.com/Kong/kong/issues/3111

The docs are a bit misleading. The docs describe the evaluation process, not the matching order. Matching order is for now: exact, regex, prefix. Until we add support for `^~`.